src/libstore/unix/build/freebsd-derivation-builder.cc

```diff
@@ -0,0 +1,501 @@
+                    // want to run resolved or some other local resolver anyway.
+                    //
+                    // There's also just no simple way to do this correctly, you have to manually
+                    // inotify watch the files for changes on the outside and update the sandbox
+                    // while the build is running (or at least that's what Flatpak does).
+                    //
+                    // I also just generally feel icky about modifying sandbox state under a build,
+                    // even though it really shouldn't be a big deal. -K900
+                    copyFile(path, std::filesystem::path{chrootRootDir} / path.relative_path(), false, true);
+                }
+            }
+
+            if (settings.caFile != "" && pathExists(std::filesystem::path{settings.caFile.get()})) {
+                // For the same reasons as above, copy the CA certificates file too.
+                // It should be even less likely to change during the build than resolv.conf.
+                createDirs(chrootRootDir + "/etc/ssl/certs");
+                copyFile(
+                    std::filesystem::path{settings.caFile.get()},
+                    chrootRootDir + "/etc/ssl/certs/ca-certificates.crt",
+                    false,
+                    true);
+            }
+        }
+    }
+
+    void startChild() override
+    {
+        int jid;
+
+        RunChildArgs args{
+#  if NIX_WITH_AWS_AUTH
+            .awsCredentials = preResolveAwsCredentials(),
+#  endif
+        };
+
+        if (derivationType.isSandboxed()) {
+            jid = jail_setv(
+                JAIL_CREATE,
+                "persist",
+                "true",
+                "path",
+                chrootRootDir.c_str(),
+                "host.hostname",
+                "localhost",
+                // TODO: Make our own ruleset
+                "vnet",
+                "new",
+                NULL);
+            if (jid < 0) {
+                throw SysError("Failed to create jail (isolated network): %1%", jail_errmsg);
+            }
+            autoDelJail = std::make_shared<AutoRemoveJail>(jid);
+
+            // Everything from here to the end of the block is setting up the network
+            AutoCloseFD fd(socket(PF_INET, SOCK_DGRAM, 0));
+            if (!fd)
+                throw SysError("cannot open IP socket");
+
+            struct ifreq ifr;
+            strcpy(ifr.ifr_name, "lo0");
+            ifr.ifr_flags = IFF_UP | IFF_LOOPBACK;
+            if (ioctl(fd.get(), SIOCSIFFLAGS, &ifr) == -1)
+                throw SysError("cannot set loopback interface flags");
+
+            AutoCloseFD netlink(socket(PF_NETLINK, SOCK_RAW, NETLINK_ROUTE));
+
+            struct
+            {
+                struct nlmsghdr nl_hdr;
+                struct ifaddrmsg addr_msg;
+                struct nlattr tl;
+                uint8_t addr[4];
```

Comment:
Alternatively this could just be a `std::array`. Standard C++ basically mandates that std::array is an aggregate type with only the C-style array as the only member.

```suggestion
                std::array<uint8_t, 4> addr;
```